android-components/components/browser/state/src/main/java/mozilla/components/browser/state/engine/middleware/TranslationsMiddleware.kt

```diff
@@ -39,8 +41,52 @@ class TranslationsMiddleware(
     ) {
         // Pre process actions
         when (action) {
+            // Initially adding the tab
```

Comment:
Maybe include covering first session load which doesn't add a new tab.

Reply:
Switched to a new pattern, but I'll remember this if I have to do something similar with initializing the tab states. (I'm going to take a deeper look at web extensions and around in general if I can find some other alternatives for the tabs too.)